packages/dds/map/src/test/memory/map.spec.ts

```diff
@@ -11,6 +14,24 @@ import {
 import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
 
 import { type ISharedMap, SharedMap } from "../../index.js";
+const baselineFilePath = path.join(__dirname, "memory_baseline.json");
+
+// Load previous baseline if exists
+const loadBaselines = (): Record<string, number> => {
+	try {
+		return JSON.parse(fs.readFileSync(baselineFilePath, "utf8")) as Record<string, number>;
+	} catch {
+		return {};
+	}
+};
+
+// Function to update baseline file after test runs
+const saveBaseline = (testTitle: string, memoryUsage: number): void => {
+	const baselines = loadBaselines();
+	baselines[testTitle] = memoryUsage;
+	// eslint-disable-next-line unicorn/no-null
+	fs.writeFileSync(baselineFilePath, JSON.stringify(baselines, null, 2));
```

Comment:
Specified the encoding as "utf8" in the fs.writeFileSync() method to ensure consistent file writing. Like: fs.writeFileSync(baselineFilePath, JSON.stringify(baselines, null, 2), "utf8");